src/components/Button/index.tsx

```diff
@@ -257,3 +263,58 @@ export function ButtonDropdownLight({
     </ButtonOutlined>
   )
 }
+
+const BtnApprovedWraper = styled(ButtonConfirmed)`
+  padding: 0;
+  height: 44px;
+  display: flex;
+  align-items: center;
+  width: 48%;
+`
+
+export const ButtonApprove = ({
+  tooltipMsg,
+  tokenSymbol,
+  approval,
+  onClick,
+  disabled,
+  forceApprove = false,
+}: {
+  tooltipMsg?: string
+  tokenSymbol: string | undefined
+  approval: ApprovalState
+  onClick: () => void
+  disabled: boolean
+  forceApprove?: boolean
+}) => {
+  return (
+    <BtnApprovedWraper
+      disabled={disabled}
+      onClick={onClick}
+      altDisabledStyle={approval === ApprovalState.PENDING} // show solid button while waiting
+      confirmed={approval === ApprovalState.APPROVED && !forceApprove}
+    >
+      {approval === ApprovalState.PENDING ? (
+        <AutoRow gap="6px" justify="center">
+          <Trans>Approving</Trans> <Loader stroke="white" />
+        </AutoRow>
+      ) : (
+        <>
+          {tooltipMsg && (
+            <MouseoverTooltip width="300px" text={tooltipMsg}>
+              <Flex
+                style={{ alignItems: 'center', height: 44, paddingRight: 8, paddingLeft: 2 }}
+                onClick={e => e.stopPropagation()}
```

Comment:
Can you help explain usecase of this `stopPropagation`? I rarely see this being used.

Reply:
on mobile, it is very hard to click tooltip here, after discuss with Kami and QC, we decide to enlarge touch region for tooltip by wrap it by a div , use stopPropagation to prevent event click button when click this div

![image](https://user-images.githubusercontent.com/33005392/206611545-eed6cd9f-72d1-406e-a2db-aed7258762e3.png)

Reply:
That's cool. But IMO we should put that tooltip outside button.

Reply:
Agree with Nam, but also agree with Danh. Product team decided that way, we should follow